Core/DailyPixel.swift

```diff
@@ -97,19 +107,27 @@ public final class DailyPixel {
         )
     }
     
-    private static func updatePixelLastFireDate(pixel: Pixel.Event) {
-        storage.set(Date(), forKey: pixel.name)
+    private static func updatePixelLastFireDate(forKey key: String) {
+        storage.set(Date(), forKey: key)
     }
-    
-}
 
-private extension Pixel.Event {
-    
-    func hasBeenFiredToday(dailyPixelStorage: UserDefaults) -> Bool {
-        if let lastFireDate = dailyPixelStorage.object(forKey: name) as? Date {
+    private static func hasBeenFiredToday(forKey key: String, dailyPixelStorage: UserDefaults) -> Bool {
+        if let lastFireDate = dailyPixelStorage.object(forKey: key) as? Date {
             return Date().isSameDay(lastFireDate)
         }
         return false
     }
-    
+
+    private static func createSortedStringOfValues(from dict: [String: String], maxLength: Int = 50) -> String {
+        let sortedKeys = dict.keys.sorted()
+
+        let uniqueString = sortedKeys.map { "\(dict[$0]!)" }.joined(separator: ";")
```

Comment:
We could probably use `compactMap` here to filter out nil values and avoid the force unwrap, but tbh I can't think of a scenario where this would fail.